### ISSUE TITLE
How many epochs used and how to calculate model error loss?

### ISSUE THREAD

**Issue by GenTxt** [ADGEfficiency/creative-writing-with-gpt2#2]
Thanks for this cool version of the Transformer repo.
Have everything working and currently testing with 345M, 774M, and 1558M on a 24 gig VM (with parameter tweaking on block_size for 1558M)
I've been using gptune and gpt2-finetune TF repos where it's easy to see real-time error loss in terminal or set the desired model loss (gptune) but I can't find a similar feature with pytorch transformers. Checked huggingface  issues but couldn't find the answer. Any suggestions are appreciated.

Also, did you use only 4 epochs as per your example to generate your 117M models? 

The quality is very good for small, single book models. I find the transformer version handles small corpus files better than TF repos which can generate significant over-fitting.

Cheers

**Comment by ADGEfficiency** [comment]
All the pre-finetuned models are with 4 epochs - this was the first I tried and I liked the results :)

I don't know how to see the real time loss in the shell.  

Thanks for the tip re single book models!
